microservices/object_detection/detector.py

```diff
@@ -1,42 +1,39 @@
 from config import PATH_TO_MODEL, THRESHOLD
 import detector_base
+from config import log
 import detect
 from annotator import Annotator
 import bottle
-from bottle import route, run,  get, post, request 
+from bottle import route, run,  get, post, request, response
 import io
 from PIL import Image
 from PIL import ImageDraw
 
 @route('/')
 def greetings():
-    return "Hi, I am the AI4Coral RESTful API\n"
-
+  return "Hi, I am the AI4Coral RESTful API\n"
 
 @post('/detect/<filename>')
 def do_detections(filename):
-    try:
-        try:    
-            interpreter = detector_base.make_interpreter(PATH_TO_MODEL)
-            interpreter.allocate_tensors()
+  try:
```

Comment:
I think one possibility for logging is to calculate the time taken to load image, and get annotation. That will give us an idea if one part is taking too much time in separate. So we can check our bottlenecks per single image on average by only using the `time` module to calculate the time difference.

I'm not fully aware of how the `logging` module work, but it would be useful to have a file which we could somehow parse with these values so we could plot or create tables to gain insights.

Reply:
got it, but can you elaborate more on that?\

I did not understand the calculation part, like, which sessions should we calculate?\ 
time need to load the image, time to do the detection, time to build the pascal voc xml, etc 

I can create a log sinilar to a csv file, so we can do the analysis later with pandas ...

Reply:
Ah, sorry, the calculation part is just how usually these times are calculated. Usually is something similar to:

```python
t0 = get_timetep()
# Do calculation 
t1 = get_timestep()

total_time_taken = t1 - t0 # this is the calculation I was mentioning.
```

So, I'd say we could make these calculations in separate for loading the image, detection, building the xml, etc etc.

If we could have a csv file with these values it would be extremely useful to facilitate plot creation and further analysis :)

Reply:
got it